iceoryx_hoofs/vocabulary/include/iox/optional.hpp

```diff
@@ -228,13 +230,15 @@ class optional final : public FunctionalInterface<optional<T>, T, void>
     //     initHandle(&handle);
     //   }
     bool m_hasValue{false};
+    // AXIVION DISABLE STYLE AutosarC++19_03-A9.6.1 : False positive. Used type has defined size.
     struct alignas(T) element_t
     {
-        // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the array
+        // AXIVION Next Construct AutosarC++19_03-A18.1.1, AutosarC++19_03-A1.1.1, AutosarC++19_03-M0.1.3 : required as low level building block, encapsulated in abstraction and not directly used
         // is wrapped inside the optional
         // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
         byte_t data[sizeof(T)];
     };
+    // AXIVION Next Construct AutosarC++19_03-A1.1.1 : object size limit is not relevant for containers stored in shared memory
```

Comment:
`iox::optional` is not only stored in shared memory, but also on the stack and some users may store them on the heap?

Reply:
@mossmaurice To your second question: in https://github.com/eclipse-iceoryx/iceoryx/pull/1749#discussion_r1004575064 we came to the conclusion `UninitializedArray<T, 1> m_data` or implementing an "UninitializedCell" would be overkill for the few remaining occurrences of these warnings.

Reply:
I plagiarized from `UninitializedArray`. Have to think of a better reasoning and maybe I remove the suppression for now and fix it in a follow up PR

Reply:
Found a better justification :)